src/main.c

```diff
@@ -646,6 +648,17 @@ static int create_socket (const char *node, const char *service) /* {{{ */
 			continue;
 		}
 
+		// Set receive timeout on the socket if one is set
+		if(NULL != timeout)
+		{
+			if (setsockopt (fd, SOL_SOCKET, SO_RCVTIMEO, (char *)timeout, sizeof(struct timeval)) < 0)
```

Comment:
Please remove the cast (or cast `timeout` to `(void *)` if casting is necessary at all). `setsockopt()` takes a `const void *` and the C compiler should not require an explicit cast to `void *`.